### PR TITLE
Don't log so many stkaing events on genesis block

### PR DESCRIPTION
**Pull request by kianenigma**
Not so pretty on the code, but these logs are polluting other commands as well that build the genesis spec. For example, running benchmarking before would give you:

```
     Running `target/release/substrate benchmark --chain=dev --steps=5 --repeat=5 --pallet=pallet_election_provider_multi_phase --extrinsic=elect_queued --execution=native --wasm-execution=compiled --heap-pages=4096 --output=.`
2021-03-12 12:50:16.591  TRACE main runtime::staking: 💸 [0] planning new_session(0)
2021-03-12 12:50:16.591   INFO main runtime::staking: 💸 new validator set of size 1 has been elected via ElectionCompute::OnChain for staring era 0
2021-03-12 12:50:16.591   WARN main runtime::election-provider: 🗳 Failed to finalize election round. reason NoFallbackConfigured
2021-03-12 12:50:16.591  DEBUG main runtime::staking: 💸 Experimental election provider outputted Err(NoFallbackConfigured)
2021-03-12 12:50:16.591  TRACE main runtime::staking: 💸 [0] planning new_session(1)
2021-03-12 12:50:16.591  TRACE main runtime::staking: 💸 [0] starting start_session(0)
```

all of which are gone now.

### CODE REVIEW

**Comment by kianenigma** [comment]
would be also good to gate 

```
2021-03-13 11:41:56.915   INFO main runtime::election-provider: 🗳 Finalized election round with compute ElectionCompute::OnChain.
2021-03-13 11:41:56.916   WARN main runtime::staking: [0] 💸 chain does not have enough staking candidates to operate for era 0
```

**Review by bkchr — APPROVED** [review]

File: frame/staking/src/lib.rs [hunk 1]
```diff
@@ -3178,7 +3182,7 @@ impl<T: Config> Module<T> {
 	/// Enact and process the election using the `ElectionProvider` type.
 	///
 	/// This will also process the election, as noted in [`process_election`].
-	fn enact_election(_current_era: EraIndex) -> Option<Vec<T::AccountId>> {
+	fn enact_election(current_era: EraIndex) -> Option<Vec<T::AccountId>> {
```
Comment (kianenigma) [inline frame/staking/src/lib.rs:3185]:
```suggestion
	fn enact_election(_current_era: EraIndex) -> Option<Vec<T::AccountId>> {
```

**Review by thiolliere — APPROVED** [review]

File: frame/staking/src/lib.rs [hunk 2]
```diff
@@ -3129,11 +3131,13 @@ impl<T: Config> Module<T> {
 		let elected_stashes = exposures.iter().cloned().map(|(x, _)| x).collect::<Vec<_>>();
 
 		if (elected_stashes.len() as u32) <= Self::minimum_validator_count() {
-			log!(
-				warn,
-				"chain does not have enough staking candidates to operate for era {:?}",
-				current_era,
-			);
+			if current_era != 0 {
```
Comment (kianenigma) [inline frame/staking/src/lib.rs:3134]:
```suggestion
			if current_era > 0 {
```

**Comment by kianenigma** [comment]
bot merge

File: frame/staking/src/lib.rs (hunk 2 above)
Comment (shawntabrizi) [inline frame/staking/src/lib.rs]: `!= 0` vs ` > 0` is the same?

so why the difference?

**Review by shawntabrizi — APPROVED** [review]

File: frame/staking/src/lib.rs (hunk 2 above)
Comment (kianenigma) [reply frame/staking/src/lib.rs]: I just updated it while you were putting the comment :P

**Pull request merged**